### ISSUE TITLE
Missing Material

### ISSUE THREAD

**Issue by RickG73** [Autodesk/revit-ifc#161]
Hello,

I have a problem in different versions of Revit and the exporter. In some cases the Material disappears in the export. We tracked it back to more than one causes (overlap of elements, a join, using a void, using edit boundry, a floor in an angle) but it's not consistent. At first we thought the problem was to be found in Solibri, but after consulting them it's obvious the IfcMaterialLayerSet of IfcMaterial isn't filled with a value. By the way, in IFC4 the problem is solved! Unfortunately, by contract we have to use 2X3.

I've attachted three files. The RVT (v19) with some floors and two export IFC's.  A 2x3 with the disappearing material and the IFC4 where the material is present!

I can only imagine that the problem lies with the exporter but I really want to hear your opinions...

Thx, Rick
[missing material.zip](https://github.com/Autodesk/revit-ifc/files/4324219/missing.material.zip)

**Comment by RickG73** [comment]
anyone?

**Comment by AngelVelezSosa** [comment]
We will file an internal problem report.  @parrela

**Comment by RickG73** [comment]
Thanks a lot.

**Comment by RustyMetal** [comment]
I have filed an item to investigate. @RickG73  are you using the latest open source exporter? if you are, what if you uninstall it and go with the default?

**Comment by RickG73** [comment]
@RustyMetal , Yes, tried in Revit 18-19-20 all versions of the exporter. I just uninstalled the exporter 20.1 and tried the default IFC exporter. Same problem unfortunately.

**Comment by RickG73** [comment]
> We will file an internal problem report. @parrela

@AngelVelezSosa , will there be a follow-up report or feedback?

**Comment by RickG73** [comment]
Are we the only ones who are having this problem? Plz let me know...

**Comment by Moult** [comment]
2 years later I'm also coming across this issue, I can't seem to find any pattern to it yet. Here is one such portion of the model, regular steel beam, with a steel material.

![image](https://user-images.githubusercontent.com/88302/200997219-2625fef3-28f8-414f-9f0f-839fcda5919e.png)

Here I've isolated all the members which are missing materials.

![image](https://user-images.githubusercontent.com/88302/200997336-e0474aff-a829-4f1c-a964-51f571f2ecc6.png)

**Comment by Moult** [comment]
I'd need to ask for permission on whether or not it can be shared, it's not our file, as we work with a lot of consultants.

**Comment by AngelVelezSosa** [comment]
Always an issue.  If you can get just the subset of the file that has the element with the missing material, that would help both of us - in general, we spend a lot of time reducing the data we get to the actually interesting bit, which is usually less than 1% of it.

**Comment by ArtemHnatko** [comment]
The issue of missing materials in IFC2x3 exports is not reproducible in Revit 2023-2025. Materials are correctly exported in both IFC2x3 and IFC4 formats.

**Testing environment:**
- Autodesk Revit 2023 (v.23.1.60.0)
- Autodesk Revit 2024
- Autodesk Revit 2025

![Image](https://github.com/user-attachments/assets/1b68dbd7-1d68-48e1-8a65-5c36f5ae57a5)
